src/coreclr/tools/r2rdump/CoreDisTools.cs

```diff
@@ -1209,6 +1213,235 @@ private static bool IsAnotherRuntimeFunctionWithinMethod(int rva, RuntimeFunctio
             return false;
         }
 
+        /// <summary>
+        /// Improves disassembler output for RiscV64 by adding comments at the end of instructions.
+        /// </summary>
+        /// <param name="rtf">Runtime function</param>
+        /// <param name="imageOffset">Offset within the image byte array</param>
+        /// <param name="rtfOffset">Offset within the runtime function</param>
+        /// <param name="instruction">Textual representation of the instruction</param>
+        private void ProbeRiscV64Quirks(RuntimeFunction rtf, int imageOffset, int rtfOffset, ref string instruction)
+        {
+            const int InstructionSize = 4;
+            uint instr = BitConverter.ToUInt32(_reader.Image, imageOffset + rtfOffset);
+
+            /*
+                Pattern:
+                    auipc
+                    addi
+                    ld
+                    jarl
```

Comment:
Update `jarl` to `jalr`.
Can you share code locations in runtime where you checked the patterns?

Reply:
There is no a strict line of code which checks this pattern. 

Function `ProbeRiscV64Quirks` is looking for `jalr` instruction. If it finds it, then it goes back through the assembly code to calculate callee address. It is more analytical approach, than the fixed pattern. Unfortunately, instructions calculating this address do not always follow each other, sometimes unrelated instruction appears between them. Other problem was with multiple-wrapped calle, then more than one `ld` or `lw` are used to calculate callee address.

Reply:
Well. you cannot give codes in runtime for the patterns.
However, in other archs, their implementation is different from yours. And it looks like they have some strict patterns.
And if this code should be checked manually without any tests, I cannot review whether this is correct implementation or not.
More, I don't know well about r2rdump. I am so sorry.

@jkotas @ivdiazsa Could you review this PR?
Thank you.

Reply:
> I don't know well about r2rdump

The point of this code in r2rdump is to make the disassembly output more human readable. Nothing in the rest of the runtime depends on it. There are no hard and fast rules for how it should behave.

@SzpejnaDawid Could you please share some examples for what the output looks like with this change, including the problematic cases like more than one ld or lw are used?

Reply:
I think that `System.Private.CoreLib.dll` has some interesting examples:
```
System.Exception Interop.GetExceptionForIoErrno(Interop+ErrorInfo, string, bool)
...
14811c: 00e00397    auipc   t2, 3584
148120: f443b383    ld      t2, -188(t2)
148124: 000380e7    jalr    t2 // <----
...
148528: 00e0cf17    auipc   t5, 3596
14852c: 550f0f13    addi    t5, t5, 1360
148530: 00000593    li      a1, 0
148534: 000f3603    ld      a2, 0(t5)
148538: 000600e7    jalr    a2 // <----
...
148840: 00e10f17    auipc   t5, 3600
148844: a20f0f13    addi    t5, t5, -1504
148848: 000f3583    ld      a1, 0(t5)
14884c: 000580e7    jalr    a1 // <----
...

int Interop+Globalization.GetCalendars(string, System.Globalization.CalendarId[], int)
...
149360: c5450513    addi    a0, a0, -940
149364: 00053503    ld      a0, 0(a0)
149368: 00053683    ld      a3, 0(a0)
14936c: fe443503    ld      a0, -28(s0)
149370: fec42603    lw      a2, -20(s0)
149374: fdc43583    ld      a1, -36(s0)
149378: fd443f03    ld      t5, -44(s0)
14937c: 000680e7    jalr    a3 // <----
```
I know that my way of calculation callee address is not fixed as for `arm`. Nevertheless, the only way to be sure how jumps are generated is to read a code of `coregen2` in order to see how it generates `riscv` instructions. But the profit and loss ratio of this approach is unfavorable, imo.